electricitymap/contrib/lib/models/event_lists.py

```diff
@@ -227,6 +230,8 @@ def merge_production_breakdowns(
 
         df = df.drop(columns=["source", "sourceType", "zoneKey"])
         df = df.groupby(level=0, dropna=False)["data"].apply(list)
+        if matching_timestamps_only:
```

Comment:
Here we make sure that the grouped events for each row match the number of list we want to merge.
This eliminates rows were a datetime is missing in the other list.